### PR TITLE
Exposing unifrac.meta

### PR DESCRIPTION
**Pull request by wasade**
This is a work in progress pull request that exposes `unifrac.meta`. Would it be possible to get a very quick comment on the approach to handling the passthrough? I wanted to make sure this approach was reasonable before getting the rest of the bits finalized. 

The TODO items that I'm aware of are:

- [x] adjust plugin_setup text to better reflect the method (currently effectively a copy/paste of `beta_phylogenetic`)
- [x] add support for `meta` specific arguments
- [x] add support for checking empty tables

@ebolyen and @thermokarst, per the discussion on handling paired inputs on Slack, I'd be happy to revise this to reflect the `:` syntax but also wanted to get something tentatively in while that discussion solidifies. I'd also be happy to submit a PR to `q2cli` to support the `:` syntax if that is considered a blocker for this PR, although I may need some guidance there as (at the moment) am not very familiar with that codebase.

### CODE REVIEW

File: q2_diversity_lib/beta.py [hunk 1]
```diff
@@ -117,6 +117,37 @@ def beta_phylogenetic_passthrough(table: BIOMV210Format,
...
+@_validate_requested_cpus
+def beta_phylogenetic_meta_passthrough(table: BIOMV210Format,
+                                       phylogeny: NewickFormat,
+                                       metric: str,
+                                       threads: int = 1,
+                                       variance_adjusted: bool = False,
+                                       alpha: float = None,
+                                       bypass_tips: bool = False,
+                                       weights: list = None,
+                                       consolidation: str='skipping_missing_values'  # noqa
```
Comment (thermokarst) [inline q2_diversity_lib/beta.py:130]: 🧙

File: q2_diversity_lib/tests/test_beta.py [hunk 2]
```diff
@@ -331,6 +332,43 @@ def test_accepted_types_have_consistent_behavior(self):
...
+    def setUp(self):
+        super().setUp()
+        self.method = self.plugin.actions['beta_phylogenetic_meta_passthrough']
+        empty_table = biom.Table(np.array([]), [], [])
+        self.empty_table = Artifact.import_data('FeatureTable[Frequency]',
+                                                empty_table)
+
+        # checking parity with the unifrac.meta tests
+        table1 = pkg_resources.resource_filename('unifrac.tests',
+                                                 'data/e1.biom')
```
Comment (thermokarst) [inline q2_diversity_lib/tests/test_beta.py:347]: Do you think we should copy these test data over to this plugin?

**Comment by thermokarst** [comment]
BTW, I sorted out the CI issues, you might need a refresh from `master`.

File: q2_diversity_lib/tests/test_beta.py (hunk 2 above)
Comment (wasade) [reply q2_diversity_lib/tests/test_beta.py:347]: No preference. This *might* be the regular crawford data actually

**Comment by wasade** [comment]
Thanks @thermokarst!!

File: ci/recipe/meta.yaml [hunk 3]
```diff
@@ -27,7 +27,7 @@ requirements:
     - q2-types {{ release }}.*
     - scikit-bio
     - scipy
-    - unifrac
+    - unifrac >=0.20.0
```
Comment (thermokarst) [inline ci/recipe/meta.yaml:30]: @wasade - what do you think of setting a minimum version for unifrac? If "yes" - is this the right version?

Comment (wasade) [reply ci/recipe/meta.yaml:30]: :+1:

**Comment by thermokarst** [comment]
Hey @wasade - I got the CI back in order, looks like there is a legit test failure. Let me know if you want me to take a look at it.

**Comment by thermokarst** [comment]
Actually, looks like maybe that is unrelated to the changeset here? @wasade, can you comment on this - do you know if the phylo/table mis-match error changed in the latest release of unifrac?

cc @ChrisKeefe

**Comment by wasade** [comment]
biocore/unifrac#106 I think? These extra validations were added

**Comment by thermokarst** [comment]
Thanks for the reminder @wasade, looks like @ChrisKeefe left some breadcrumbs here - I'll push a commit up to this feature branch sometime soon to address.

**Comment by thermokarst** [comment]
https://github.com/qiime2/q2-diversity-lib/blob/5c277ae800c002a31b01de19c48a22cfe621c217/q2_diversity_lib/tests/test_beta.py#L69-L72

**Review by thermokarst — APPROVED** [review]
🎉

**Comment by wasade** [comment]
Excellent!!

**Pull request merged**